shark_turbine/kernel/wave/promotion.py

```diff
@@ -0,0 +1,41 @@
+from ...support.logging import get_logger
+from shark_turbine.kernel._support.tracing import CapturedTrace
+import torch.fx as fx
+from ..ops.wave_ops import *
+from .address_spaces import *
+import shark_turbine.kernel.lang as tkl
+
+logger = get_logger("turbine.wave.promotion")
+
+
+def apply_promotion_pattern_(custom_node: Read | Write, allocate_node: Allocate):
+    match custom_node:
+        case Read(
```

Comment:
We could add a `case _: ` to report that we encountered a node we do not know how to do promotion for.
Is it correct that we expect `custom_node: Read | Write` but do not handle the `Write` case in the match?

Reply:
Yes, that's correct. Writes can also be promoted, I just haven't implemented that yet. I will add a case to handle that.